athena-sqlserver/src/main/java/com/amazonaws/athena/connectors/sqlserver/SqlServerMetadataHandler.java

```diff
@@ -208,7 +208,7 @@ public void getPartitions(BlockWriter blockWriter, GetTableLayoutRequest getTabl
                     // Include the first partition because it's not retrieved from GET_PARTITIONS_QUERY
                     blockWriter.writeRows((Block block, int rowNum) ->
                     {
-                        block.setValue(PARTITION_NUMBER, rowNum, "1");
+                        block.setValue(PARTITION_NUMBER, rowNum, "1" + ":::" + partitionFunction + ":::" + partitioningColumn);
```

Comment:
`partitionFunction` and `partitioningColumn` are only ever used inside this function, so there is no need for them to be class level member variables.

Update `getPartitionFunction` to return these values rather than having it set them as class level member variables.

Additionally, what happens if `getPartitionFunction` doesn't set the value of the `partitionFunction` or `partitioningColumn` variables? (meaning when `resultSet.next()` on line 302 doesn't return true).

If this is a possible scenario, add a test to cover this scenario in addition to fixing the code to handle this scenario.

Reply:
@henrymai 
I got your point, there will not be case wherein the resultSet.next() become false. I have tested the code thoroughly.
But if we get such scenario will analyze it and required changes will be done.